components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -127,6 +132,22 @@ public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx)
             }
             // Return a new access token in each request when JWTTokenIssuer is used.
             if (accessTokenNotRenewedPerRequest(oauthTokenIssuer, tokReqMsgCtx)) {
```

Comment:
We might need to document these scenarios properly. By looking at this code, it seems like there are multiple cases where new access token can be issued. To me accessTokenNotRenewedPerRequest and isTokenRenewalPerRequestEnabled are confusing.

Reply:
accessTokenNotRenewedPerRequest is used when JWTTokenIssuer is used and only renews the access token. isTokenRenewalPerRequestEnabled will actually revoke the existing access token and refresh token so a new token pair will be generated.

Reply:
accessTokenNotRenewedPerRequest is not just JWTTokenIssuer specific it seems. It's something which is coming from token issuer. Do we have these documented in a developer guide or somewhere ?

Reply:
Mentioned JWTTokenIssuer only in the previous comment as only that class has overridden the renewAccessTokenPerRequest method from the interface and sends "true". For others the default value false is sent. If someone uses the [OAuthTokenGenerator extension point ](https://docs.wso2.com/display/IS570/Extension+Points+for+OAuth#ExtensionPointsforOAuth-OAuthTokenGenerator) and overrides the renewAccessTokenPerRequest methods to send "true" then isTokenRenewalPerRequestEnabled will not be used.  Not sure on the documentation, will check on how we can document this clearly.